### PR TITLE
Make platform validation optional; re-enable for all analyses except geneprint

### PR DESCRIPTION
**Pull request by kretep**

### CODE REVIEW

File: grails-app/views/plugin/Geneprint.gsp [hunk 1]
```diff
@@ -19,7 +19,7 @@
                 <span>Select a High Dimensional Data node from the Data Set Explorer Tree and drag it into the box.</span>
                 <div id='divIndependentVariable' class="queryGroupIncludeSmall highDimBox"></div>
                 <div class="highDimBtns">
-                    <button type="button" onclick="highDimensionalData.gather_high_dimensional_data('divIndependentVariable')">High Dimensional Data</button>
+                    <button type="button" onclick="highDimensionalData.gather_high_dimensional_data('divIndependentVariable', false, false)">High Dimensional Data</button>
```
Comment (forus) [inline grails-app/views/plugin/Geneprint.gsp]: What checks are still active for geneprint? Do you need any HD node validation at first place?

File: web-app/js/HighDimensionalData.js [hunk 2]
```diff
@@ -350,10 +350,12 @@ HighDimensionalData.prototype.get_inputs = function (divId) {
     ]
 }
 
-HighDimensionalData.prototype.gather_high_dimensional_data = function (divId, hideAggregration) {
+HighDimensionalData.prototype.gather_high_dimensional_data = function (divId, hideAggregration, doValidatePlatforms) {
```
Comment (forus) [inline web-app/js/HighDimensionalData.js]: I do not see any call of this method with `doValidatePlatforms = true`. Are you sure you didn't disable platform validation for all analyses?

File: web-app/js/HighDimensionalData.js [hunk 3]
```diff
@@ -394,16 +396,19 @@ HighDimensionalData.prototype.gather_high_dimensional_data = function (divId, hi
...
+          var formValidator = new FormValidator(platforms);
+
+          if (formValidator.validateInputForm()) {
+            _this.display_high_dimensional_popup();
+          } else {
+            formValidator.display_errors();
+          }
+        }
+        else {
+          _this.display_high_dimensional_popup();
```
Comment (forus) [inline web-app/js/HighDimensionalData.js]: Overall condition could be simplified. `display_high_dimensional_popup() ` could be called once when there are no errors.

File: grails-app/views/plugin/Geneprint.gsp (hunk 1 above)
Comment (kretep) [inline grails-app/views/plugin/Geneprint.gsp]: There are two parameters being passed here, the first of which is false (or undefined) by default. The second is true by default, so only for geneprint it is explicitly set to false.

File: web-app/js/HighDimensionalData.js (hunk 2 above)
Comment (kretep) [inline web-app/js/HighDimensionalData.js]: The value is true by default if it is not specified. See the third line in this method.

File: web-app/js/HighDimensionalData.js (hunk 3 above)
Comment (kretep) [inline web-app/js/HighDimensionalData.js]: I'm not sure what you mean. It's only called once, right? Once if validation succeeds or once if no validation is required.

**Pull request merged**